builtins: block gateway_region from DistSQL

otan opened the pull request:
Resolves: #69598

Release justification: safety change

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69624)
<!-- Reviewable:end -->

otan committed:
builtins: block gateway_region from DistSQL

It is not safe to execute this in DistSQL as we always want the region
of the gateway to be the return value here.

Release justification: safety change

Release note: None

otan commented:
bors r=arulajmani
